crates/biome_js_analyze/src/lint/nursery/use_consistent_member_accessibility.rs

```diff
@@ -0,0 +1,558 @@
+    ///   }
+    ///   walk() {
+    ///     // method
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ## Options
+    ///
+    /// The rule supports the following options:
+    ///
+    /// ```json
+    /// {
+    ///     "//": "...",
+    ///     "options": {
+    ///         "accessibility": "explicit"
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### `accessibility`
+    /// This option determines the required accessibility modifiers on class properties and methods.
+    /// It can be set to one of the following values:
+    /// - `noPublic` - forbid the use of public (a safe fix will remove it).
+    /// - `explicit` - requires an accessibility modifier for every member that allows that (a safe fix will add public).
+    /// - `none` - forbid all accessibility modifiers (public, protected, private).
+    ///
+    /// Default: `noPublic`.
+    ///
+    pub UseConsistentMemberAccessibility {
+        version: "1.8.0",
+        name: "useConsistentMemberAccessibility",
+        language: "ts",
+        sources: &[RuleSource::Eslint("explicit-member-accessibility")],
+        recommended: false,
+        source_kind: RuleSourceKind::Inspired,
+    }
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct UseConsistentMemberAccessibilityOptions {
+    accessibility: Accessibility,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub enum Accessibility {
+    #[default]
+    NoPublic,
+    Explicit,
+    None,
+}
+
+impl Rule for UseConsistentMemberAccessibility {
+    type Query = Ast<AnyMember>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = UseConsistentMemberAccessibilityOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        match ctx.options().accessibility {
+            Accessibility::NoPublic => {
+                if node.is_no_public() == Some(false) {
+                    Some(())
+                } else {
+                    None
+                }
+            }
+            Accessibility::Explicit => {
+                if node.is_explicit() == Some(false) {
+                    Some(())
+                } else {
+                    None
+                }
+            }
+            Accessibility::None => {
+                if node.is_none() == Some(false) {
+                    Some(())
+                } else {
+                    None
+                }
+            }
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        match ctx.options().accessibility {
```

Comment:
could you make the variable(ex: let mut diag = RuleDiagnostic::new(// add common )) and make this code more DRY?
You can refer [here](https://github.com/biomejs/biome/blob/main/crates/biome_js_analyze/src/lint/style/use_const.rs#L112-L137) :)